pkg/utils/k8sutils.go

```diff
@@ -101,3 +106,17 @@ func AddAnnotation(obj metav1.Object, key string, value string) bool {
 	}
 	return false
 }
+
+func MapCRDAvailability(ctx context.Context, clnt client.Client, log logr.Logger, crdNames ...string) map[string]bool {
+	crdExist := map[string]bool{}
+	for _, crdName := range crdNames {
+		crd := &apiextensions.CustomResourceDefinition{}
+		crd.Name = crdName
+		if err := clnt.Get(ctx, client.ObjectKeyFromObject(crd), crd); err != nil && !errors.IsNotFound(err) {
+			log.Error(err, fmt.Sprintf("Error getting CRD  for %s", crdName))
+			os.Exit(1)
```

Comment:
There should be no reason to exit the process from a util func. 
Return an error instead and exit in the call site if needed.